api.py fix

kl290 opened the pull request:
- Funktionen korrigiert
- Tests erstellt, um Funktionalitäten abzudecken

kl290 committed:
api.py fix

- Funktionen korrigiert
- Tests erstellt, um Funktionalitäten abzudecken

kl290 committed:
api.py feature

- Kategorie-Feld hinzugefügt
- Kategorie-Validierung(vordefinierte Kategorien (health, general, work, social) erlaubt

kl290 commented:
Test

mf75 commented on api.py:
```diff
@@ -4,29 +4,55 @@
     global next_id
     data = request.get_json()
-    title, start, end = extract_data_fields(data)
+    title, start, end, category = extract_and_validate_data_fields(data)
```
ich fände `validate_and_extract_...` besser, weil es die Reihenfolge ist, in der es passiert

kl290 committed:
api.py korrekte Zeit Verwaltung

- Funktion shift_appointment erstellt
- Tests ergänzt und angepasst, um die neuen Änderungen abzudecken

kl290 committed:
api.py Weiterentwicklung von shift_appointment

- amount_start und amount_end hinzugefügt für genauere Anpassungen von Terminen
- Prüfung "Shift would cause overlapping appointment" eingebaut
- sowie Testergänzungen und Anpassungen

mf75 commented on api.py:
```diff
@@ -1,59 +1,150 @@
 appointments = []
 next_id = 1
+CATEGORY_TYPES = ["health", "general", "work", "social"]
+TIME_FORMAT = "%Y-%m-%d %H:%M"
```
Du brauchst noch einen Umgang mit dieser Forderung und ein paar Tests für das Format. 
Wenn Du von jemandem einen ungültigen String als start/end eingeliefert bekommst, der nicht zu diesem Format passt, knallt es leider und es wird ein HTTP 500 draus.

kl290 committed:
api.py Verbesserung der Fehlerbehandlung
- Umgang mit TIME_FORMAT = "%Y-%m-%d %H:%M" besser behandelt
- sowie Fehlerfälle bei POST und PUT abgefangen
